File: Generatorpassword.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"strings"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	reader := bufio.NewReader(os.Stdin)

	// Наборы символов
	app := []rune("0123456789")
	bad := []rune("abcdefghijklmnopqrstuvwxyz")
	cap := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
	den := []rune("!@#№$%^&?*_=+")

	guf := [][]rune{app, bad, cap, den}
	fuck := [][]rune{app, bad, cap}

	// Ввод длины пароля
	var length int
	for {
		fmt.Print("Введи длину пароля: ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		n, err := strconv.Atoi(input)
		if err == nil && n > 0 {
			length = n
			break
		}
		fmt.Println("Ошибка! Введите целое число")
	}

	// Нужно ли использовать спец символы
	var useSpecial bool
	fmt.Println("Надо использовать спец символы? Такие как '!#$@%&*?^_+=-'")
	for {
		fmt.Print("> ")
		answer, _ := reader.ReadString('\n')
		answer = strings.ToLower(strings.TrimSpace(answer))
		if answer == "да" {
			useSpecial = true
			break
		} else if answer == "нет" {
			useSpecial = false
			break
		} else {
			fmt.Println("Напиши да или нет")
		}
	}

	// Генерация пароля
	password := make([]rune, 0, length)
	charSets := guf
	if !useSpecial {
		charSets = fuck
	}

	for i := 0; i < length; i++ {
		set := charSets[rand.Intn(len(charSets))]
		char := set[rand.Intn(len(set))]
		password = append(password, char)
	}

	// Вывод пароля
	fmt.Println("Сгенерированный пароль:")
	fmt.Println(string(password))

	// Пауза перед выходом
	fmt.Println("\nНажмите ENTER для выхода.")
	reader.ReadString('\n')
}

File: Generatorpassword.py
from random import *
app = ["0","1","2","3","4","5","6","7","8","9",]
bad = ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"]
cap = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z"]
den = ["!","@","#","№","$","%","^","&","?","*","_","=", "+"]

guf = [app,bad,cap, den]
fuck = [app,bad,cap]

g = False
while g != True:
    try:
        y = int(input("Введи длину пароля: "))
        g = True
    except ValueError:
        print('Ошибка! Введите целое число')


print("Надо использовать спец символы? Такие как '!#$@%&*?^_+=-'")
key2 = None
key3 = None
while key2 != True:
    answer = input()
    if answer.lower() != "да" and answer.lower() != "нет":
        print("Напиши да или нет")
    else:
        if answer.lower() == "да":
            key3 = True
            key2 = True
        else:
            key2 = True
            key3 = False
password = []
if key3 == True:
    for _ in range(y):
        k  = randint(0, 3)
        i = len(guf[k])-1
        j = randint(0, i)
        password.append(guf[k][j])
else:
    for _ in range(y):
        k = randint(0, 2)
        i = len(fuck[k]) -1
        j = randint(0, i)
        password.append(fuck[k][j])
print(*password, sep="")
input()